metrics/web/lib/common/presentation/widgets/metrics_text_form_field.dart

```diff
@@ -0,0 +1,159 @@
+import 'package:flutter/material.dart';
+import 'package:metrics/common/presentation/metrics_theme/model/metrics_theme_data.dart';
+import 'package:metrics/common/presentation/metrics_theme/widgets/metrics_theme.dart';
+
+/// A widget that displays a metrics styled text field and
+/// applies the [MetricsThemeData.textFieldTheme].
+class MetricsTextFormField extends StatefulWidget {
+  /// A text field controller.
+  final TextEditingController controller;
+
+  /// A text field form validator.
+  final FormFieldValidator<String> validator;
+
+  /// An icon that appears before the editable part of this text field,
+  /// within the decoration's container.
+  final Widget prefixIcon;
+
+  /// An icon that appears after the editable part of this text field,
+  /// within the decoration's container.
+  final Widget suffixIcon;
+
+  /// Text that suggests what sort of input this field accepts.
+  final String hint;
+
+  /// Text that describes this input field.
+  final String label;
+
+  /// Creates a new instance of the [MetricsTextFormField].
```

Comment:
```suggestion
  /// Creates a new instance of the Metrics text form field.
```
Not sure if we should reference `this` widget here, I think usually it's just plain text.